app/lib/efile/id/id40_calculator.rb

```diff
@@ -13,13 +13,21 @@ def calculate
         set_line(:ID40_LINE_6C, :calculate_line_6c)
         set_line(:ID40_LINE_6D, :calculate_line_6d)
         set_line(:ID40_LINE_29, :calculate_line_29)
+
+        set_line(:ID40_LINE_43_WORKSHEET, :calculate_grocery_credit)
+        set_line(:ID40_LINE_43_DONATE, :calculate_line_43_donate)
+        set_line(:ID40_LINE_43, :calculate_line_43)
         @lines.transform_values(&:value)
       end
 
       def refund_or_owed_amount
         0
       end
 
+      def grocery_credit_amount
```

Comment:
is this just to make the method not private? what is the reason for the additional method?

Reply:
if so, i would say, wouldn't it be better to have the method be public instead so there is less obfuscation of logic?

Reply:
(discussed on tuple - this is so that we can show the amount in the title of the review page, and we don't allow calling the individual `calculate_x` methods because they frequently rely on the values of lines calculated above them)

changed this to assume that calculate has already been called, avoiding the possible bug described above